pydantic_ai_slim/pydantic_ai/format_examples.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations as _annotations
+
+import xml.etree.ElementTree as ET
+from collections.abc import Iterable
+from dataclasses import asdict, is_dataclass
+from typing import Any, Literal
+
+from pydantic import BaseModel
+
+__all__ = [
+    'format_examples',
+]
+
+
+def _to_xml_element(example: dict[str, Any]) -> ET.Element:
+    """Convert example to an xml element."""
+    element = ET.Element('example')
+    for sub_key, sub_value in example.items():
+        sub_element = ET.Element(sub_key)
+        sub_element.text = str(sub_value)
```

Comment:
we need to support further levels of nested, this should probably call `_to_xml_element`.

Reply:
we probably want one function that tries to serialize the following types to xml:
* string
* bool / int / float / None
* Sequence / Iterable
* mapping
* dataclass / pydantic model

Then we call that recursively

Reply:
Added a function _serialize_to_xml that recursively calls itself for all the datatypes specified above and converts them into 
XML elements.